File: metodos/puntoFijo.py
from math import *
import numpy as np

def puntoFijo(f, g, x0, tol, nmax):
    iter = 0
    Err = 9999
    xant = x0
    flambda = lambda x: eval(f)
    glambda = lambda x: eval(g)

    while Err> tol and iter< nmax:
        xact = glambda(xant)
        Err  = abs(xant-xact)
        print(Err)
        iter+=1
        xant = xact


    print(f"x= {xact}")
    print(f"iters = {iter}")
    print(f"Error = {Err}")


# Ejemplo de como utilizar
# puntofijo("(e**-x)-x","(e**-x)",1,5*10**(-6),12)

File: metodos/MetodosJava/PuntoFijo.java
import java.util.function.Function;

public class PuntoFijo {
    public static void puntoFijo(Function<Double, Double> f, Function<Double, Double> g, double x0, double tol, int nmax) {
        int iter = 0;
        double err = 9999;
        double xant = x0;

        while (err > tol && iter < nmax) {
            double xact = g.apply(xant);
            err = Math.abs(xant - xact);
            System.out.println(err);
            iter++;
            xant = xact;
        }

        System.out.println("x = " + xant);
        System.out.println("iters = " + iter);
        System.out.println("Error = " + err);
    }
}
